### PR TITLE
Add user tests

### PR DESCRIPTION
**Pull request by simionescuraluca**

### CODE REVIEW

File: src/test/java/com/vinyl/AddUserTest.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    private BCryptPasswordEncoder originalPasswordEncoder;
+
+    private BCryptPasswordEncoder mockedPasswordEncoder;
+
+    UserDTO userDTO = new UserDTO();
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        setUserDTO(userDTO);
```
Comment (dragosslujeru) [inline src/test/java/com/vinyl/AddUserTest.java]: do the new UserDTO() in the method and return the dto, do the assignment here, also rename method to createUserDTO()

File: src/test/java/com/vinyl/AddUserTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        Assertions.assertThat(savedUser).isNotNull();
+        Assertions.assertThat(savedUser.getFirstName()).isEqualTo(userDTO.getFirstName());
+        Assertions.assertThat(savedUser.getSecondName()).isEqualTo(userDTO.getSecondName());
+        Assertions.assertThat(savedUser.getPass()).isEqualTo(userDTO.getPass());
+    }
+
+    @Test
+    public void testWhenFirstNameIsInvalid() {
+        userDTO.setFirstName("invalid_first_name");
+        assertResponseWhenInvalidInput();
```
Comment (dragosslujeru) [inline src/test/java/com/vinyl/AddUserTest.java]: you also send the request in this method, u don't just assert

File: src/test/java/com/vinyl/AddUserTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    @Override
+    public void tearDown() {
+        passwordEncoder.setPasswordEncoder(originalPasswordEncoder);
+        super.tearDown();
+    }
+
+    @Test
+    public void testWhenOk() {
+        ResponseEntity<User> response = trt.exchange("/users", HttpMethod.POST, new HttpEntity<>(userDTO), User.class);
```
Comment (dragosslujeru) [inline src/test/java/com/vinyl/AddUserTest.java]: this line is duplicated here and in assertResponseWhenInvalid

File: src/test/java/com/vinyl/AddUserTest.java (hunk 1 above)
Comment (simionescuraluca) [reply src/test/java/com/vinyl/AddUserTest.java]: done

File: src/test/java/com/vinyl/AddUserTest.java (hunk 2 above)
Comment (simionescuraluca) [reply src/test/java/com/vinyl/AddUserTest.java]: fixed

File: src/test/java/com/vinyl/AddUserTest.java (hunk 3 above)
Comment (simionescuraluca) [reply src/test/java/com/vinyl/AddUserTest.java]: fixed

File: src/main/java/com/vinyl/service/validation/AddUserValidator.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    @Override
+    public void validate(UserDTO userDto) {
+        validateName(userDto);
+        validatePassword(userDto);
+        validateEmail(userDto);
+    }
+
+    public void validateName(UserDTO userDto) {
```
Comment (simionescuraluca) [inline src/main/java/com/vinyl/service/validation/AddUserValidator.java]: private

Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/validation/AddUserValidator.java]: done

**Review by dragosslujeru — APPROVED** [review]

**Pull request merged**